gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -136,6 +140,12 @@ public <V> void onRetry(Attempt<V> attempt) {
             }
           }
         }));
+
+    if (instrumentationEnabled && ConfigUtils.getBoolean(config, GaaSObservabilityEventProducer.GAAS_OBSERVABILITY_EVENT_ENABLED, false)) {
+      this.eventProducer = Optional.of(GaaSObservabilityEventProducer.getEventProducer(config,  jobIssueEventHandler.getIssueRepository()));
```

Comment:
slightly unconventional to construct within from a class name, rather than to make the event producer a ctor param.  using reflection internally makes the tests more challenging to write, as it appears you've found.  of course, there's a place for reflection--just at a higher level than this class.

I do see how you need the issue repo from the issue handler to create the observability event producer.  how about whereever that handler gets created and injected, that's where you'd inject an optional event producer?